src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -41,13 +43,74 @@ namespace ts.server.typingsInstaller {
         onRequestCompleted: RequestCompletedAction;
     }
 
+    interface TypeDeclarationTimestampFile {
+        entries: MapLike<number>;
+    }
+
+    function loadTypeDeclarationTimestampFile(typeDeclarationTimestampFilePath: string, host: InstallTypingHost, log: Log): MapLike<number> {
+        const fileExists = host.fileExists(typeDeclarationTimestampFilePath);
+        if (!fileExists) {
+            if (log.isEnabled()) {
+                log.writeLine(`Type declaration timestamp file '${typeDeclarationTimestampFilePath}' does not exist`);
+            }
+        }
+        try {
+            if (fileExists) {
+                const content = <TypeDeclarationTimestampFile>JSON.parse(host.readFile(typeDeclarationTimestampFilePath));
+                return content.entries || {};
+            }
+            else {
+                host.writeFile(typeDeclarationTimestampFilePath, "{}");
+                if (log.isEnabled()) {
+                    log.writeLine("Type declaration timestamp file was created.");
+                }
+                return {};
+            }
+        }
+        catch (e) {
+            if (log.isEnabled()) {
+                log.writeLine(`Error when loading type declaration timestamp file '${typeDeclarationTimestampFilePath}': ${(<Error>e).message}, ${(<Error>e).stack}`);
+            }
+            return {};
+        }
+    }
+
+    function writeTypeDeclarationTimestampFile(typeDeclarationTimestampFilePath: string, newContents: TypeDeclarationTimestampFile, host: InstallTypingHost, log: Log): void {
+        const fileExists = host.fileExists(typeDeclarationTimestampFilePath);

```

Comment:
This whole method seems to do too much; it should just be (simplified) `try { log('writing"); write(); } catch (e) { log("error"); }`

Reply:
Refactored.